From 6493c898a6e7e7dde240da31cff448363f6c33e9 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Tue, 12 Jun 2018 15:27:06 +0800
Subject: [PATCH 3964/5242] MLK-18576-2 arm64: fsl-imx8dx.dtsi: Add aux props
 for LDB nodes

commit  00c2cc199b5a6e1b91dd6f7f314668c2596ce864 from
https://source.codeaurora.org/external/imx/linux-imx.git

i.MX8dx/dxp/qxp use two LDBs(one primary, one auxiliary) to support
dual channel mode.  This patch adds DT properties in the LDB nodes
so that HWs needed by dual channel mode can be supplied.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi |   26 +++++++++++++++----------
 1 file changed, 16 insertions(+), 10 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
index 05fa892..df2d716 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
@@ -1790,18 +1790,21 @@
 		#size-cells = <0>;
 		compatible = "fsl,imx8qxp-ldb";
 		clocks = <&clk IMX8QXP_MIPI0_LVDS_PIXEL_CLK>,
-			 <&clk IMX8QXP_MIPI0_LVDS_BYPASS_CLK>;
-		clock-names = "pixel", "bypass";
-		power-domains = <&pd_mipi_dsi_0_lvds>;
+			 <&clk IMX8QXP_MIPI0_LVDS_BYPASS_CLK>,
+			 <&clk IMX8QXP_MIPI1_LVDS_PIXEL_CLK>,
+			 <&clk IMX8QXP_MIPI1_LVDS_BYPASS_CLK>;
+		clock-names = "pixel", "bypass", "aux_pixel", "aux_bypass";
+		power-domains = <&pd_mipi_dsi_0_lvds>, <&pd_mipi_dsi_1_lvds>;
 		gpr = <&lvds_region1>;
+		aux-gpr = <&lvds_region2>;
 		status = "disabled";
 
 		lvds-channel@0 {
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <0>;
-			phys = <&ldb1_phy>;
-			phy-names = "ldb_phy";
+			phys = <&ldb1_phy>, <&ldb2_phy>;
+			phy-names = "ldb_phy", "aux_ldb_phy";
 			status = "disabled";
 
 			port@0 {
@@ -1951,18 +1954,21 @@
 		#size-cells = <0>;
 		compatible = "fsl,imx8qxp-ldb";
 		clocks = <&clk IMX8QXP_MIPI1_LVDS_PIXEL_CLK>,
-			 <&clk IMX8QXP_MIPI1_LVDS_BYPASS_CLK>;
-		clock-names = "pixel", "bypass";
-		power-domains = <&pd_mipi_dsi_1_lvds>;
+			 <&clk IMX8QXP_MIPI1_LVDS_BYPASS_CLK>,
+			 <&clk IMX8QXP_MIPI0_LVDS_PIXEL_CLK>,
+			 <&clk IMX8QXP_MIPI0_LVDS_BYPASS_CLK>;
+		clock-names = "pixel", "bypass", "aux_pixel", "aux_bypass";
+		power-domains = <&pd_mipi_dsi_1_lvds>, <&pd_mipi_dsi_0_lvds>;
 		gpr = <&lvds_region2>;
+		aux-gpr = <&lvds_region1>;
 		status = "disabled";
 
 		lvds-channel@0 {
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <0>;
-			phys = <&ldb2_phy>;
-			phy-names = "ldb_phy";
+			phys = <&ldb2_phy>, <&ldb1_phy>;
+			phy-names = "ldb_phy", "aux_ldb_phy";
 			status = "disabled";
 
 			port@0 {
-- 
1.7.9.5

